conn_test.go

```diff
@@ -435,6 +444,79 @@ func TestQueryMultinodeWithMetrics(t *testing.T) {
 
 }
 
+type testRetryPolicy struct {
+	NumRetries int
+}
+
+func (t *testRetryPolicy) Attempt(qry RetryableQuery) bool {
+	return qry.Attempts() <= t.NumRetries
+}
+func (t *testRetryPolicy) GetRetryType(err error) RetryType {
+	return Retry
+}
+
```

Comment:
Test looks good. It'd be great if we could have:
* A test where the context gets done
* A test with delay zero (I'd like this to run with `-race`),which I believe is valid use case

Reply:
Sure, I'll have a go at adding that.